### ISSUE TITLE
(Regression) Gamepad non-responsive after creating / loading save states

### ISSUE THREAD

**Issue by orbea** [GLupeN64/GLupeN64#157]
OS: `Slackware64-current`
`RetroArch-2017.01.09_338980ef2_master-x86_64-1_git`
`GLupeN64-2017.01.09_799d362_master-x86_64-1_git`

When creating or loading a savestate inside a game my gamepad (Horipad one) will be non-responsive for several seconds. I have reproduced this in both Ocarina of Time and Winback. This is also not a problem with upstream, mupen64plus-libretro or snes9x-libretro and I was able to bisect it to the following commit.
```
2f21040d7658ef70ec95d46c33ec97217451a6eb is the first bad commit
commit 2f21040d7658ef70ec95d46c33ec97217451a6eb
Author: Logan McNaughton <logan@bacoosta.com>
Date:   Sat Nov 5 08:35:46 2016 -0600

    Use savestate code from upstream

:100644 100644 7bdede0592aaa56c823eeb54553a22ae399ca203 d269ccd4ca4ae4462f7ed44547600eb4b97c1e0d M	Makefile
:100644 100644 5360a51a6edf725095ce63e39c16605fba1354fd ce5f6d8ce4f82371f0d9ca7b3254c55ccd45efaf M	Makefile.common
:040000 040000 9c43ec812e6a21640df40b56eafc496d951ed4f3 e77e78183ce9aed56e6c87f154404651e803e167 M	custom
:040000 040000 1625f3f6a1dff8ac18cad1a814a19c796d378bd4 82c33a04e7bc3e6b57fc539ec6448a29c90d5cd0 M	libretro
:040000 040000 c85af048af280aa4db572bf25437b5cad335bb29 d481eb8125844839ef7506036a2d81ff997074da M	mupen64plus-core
```
2f21040d7658ef70ec95d46c33ec97217451a6eb

**Comment by orbea** [comment]
Strangely enough this is just this one usb slot....I think I will just not use it and close this. @loganmc10 if you want to pursue this more feel free to open it again, now that its revealed to be yet another problem with the usb ports on my mobo I'm happy to blame the hardware....

**Comment by orbea** [comment]
I was wrong, its this gamepad. Unplugging it and plugging it back must of caused it to work for a little bit... Potentially it could be this gamepad as it was fixed recently in the xpad driver.

https://github.com/paroj/xpad/commit/bd4f51797c2a518be90b470d499059dc088ddbd3

**Comment by orbea** [comment]
dmesg
```
[532272.290427] usb 8-1.3: new full-speed USB device number 14 using xhci_hcd
[532272.373985] usb 8-1.3: New USB device found, idVendor=0f0d, idProduct=0067
[532272.373989] usb 8-1.3: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[532272.373991] usb 8-1.3: Product: HORIPAD ONE
[532272.373993] usb 8-1.3: Manufacturer: HORI CO.,LTD.
[532272.373994] usb 8-1.3: SerialNumber: 00005DC471C67F48
[532272.376698] input: HORIPAD ONE as /devices/pci0000:00/0000:00:09.0/0000:02:00.0/usb8/8-1/8-1.3/8-1.3:1.0/input/input39
[532272.377102] xpad 8-1.3:1.0: xpad_prepare_next_out_packet - found pending output packet 0
[532272.381852] xpad 8-1.3:1.0: xpad_prepare_next_out_packet - found pending output packet 0
```

**Comment by loganmc10** [comment]
I won't lie, I'm a little reluctant to modify the savestate code in order to fix an issue with what appears to be a single controller.

It would require several modifications to upstream mupen64plus-core, since writing the save state to a file is pretty "baked in" to how the save state code works there. That is assuming that the issue really is how long the save states take to generate, and not some other random difference between the mupen64plus-libretro savestate code (that it was using), and the upstream savestate code (that it is using now).

**Comment by orbea** [comment]
That is fair enough, it might be best if this was fixed xpad if this is really a driver problem, I would prefer to keep this open at least until they have a chance to reply.

**Comment by loganmc10** [comment]
ya thats no problem, you can keep it open until it's somehow resolved

**Comment by orbea** [comment]
@loganmc10 One interesting thing I failed to notice earlier was that this will only happen with `Save State` and `Load State` in the RetroArch quick menu. It works fine with the default keybinds, `F2` (Save state) and `F4` (Load state). It still holds true that this only happens with this one gamepad though.

**Comment by orbea** [comment]
There is one more interesting behavior about this I have noticed. After the gamepad becomes non-responsive the home button will still work and the gamepad will still work fine in the RetroArch menu, but as soon as I resume the game the gamepad is still non-responsive.

My current hypothesis is that there is something in RetroArch that your innocuous change has problems with under certain use cases (My gamepad for instance...).

**Comment by loganmc10** [comment]
@orbea this should hopefully be fixed by 4fc82bfc9f72e213eff988f1d9107081e9647910

**Comment by orbea** [comment]
Yes it is, thanks!
